pybossa/auth/task.py

```diff
@@ -51,10 +51,29 @@ def _delete(self, user, task):
             return False
         return self._only_admin_or_subadminowners(user, task)
 
-    def _only_admin_or_subadminowners(self, user, task):
-        if not user.is_anonymous:
+    def _only_admin_or_subadminowners(self, user, task=None, project_owners_ids=None):
+        if not user:
+            return False
+        if user.is_anonymous:
+            return False
+        if project_owners_ids is None:
             project = self.project_repo.get(task.project_id)
```

Comment:
will this fail when both task and project_owner_ids are None?

Reply:
Yes, it will raise an exception. I'll return `False` in that case.